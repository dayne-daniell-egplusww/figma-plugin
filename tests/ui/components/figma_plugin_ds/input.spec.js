import { shallowMount } from '@vue/test-utils';
import FigmaInput from '~/ui/components/figma_plugin_ds/input.vue';
import FigmaIcon from '~/ui/components/figma_plugin_ds/icon';

describe('Figma input component', () => {
  let wrapper;
  const findInput = () => wrapper.find('input');

  function createComponent({ propsData = {} } = {}) {
    wrapper = shallowMount(FigmaInput, {
      propsData,
    });
  }

  describe('renders', () => {
    it('default component', () => {
      createComponent();
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('component with props', () => {
      createComponent({ propsData: { type: 'password', placeholder: 'Test placeholder' } });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('emits event when value changes', () => {
      createComponent();
      const inputElem = findInput();
      inputElem.element.value = 'test';
      inputElem.trigger('input');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.emitted().input[0]).toEqual(['test']);
      });
    });

    describe('when :state is false', () => {
      it('should have invalid class', () => {
        createComponent({ propsData: { state: false } });

        expect(wrapper.classes()).toContain(`input--invalid`);
      });
    });

    describe('with icon', () => {
      it('renders icon', () => {
        createComponent({ propsData: { icon: 'search' } });

        expect(wrapper.findComponent(FigmaIcon).exists()).toBe(true);
      });
    });
  });
});
