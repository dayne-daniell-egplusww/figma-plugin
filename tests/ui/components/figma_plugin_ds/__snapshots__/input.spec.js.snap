// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figma input component renders component with props 1`] = `
"<div class=\\"input\\">
  <!----> <input type=\\"password\\" placeholder=\\"Test placeholder\\" class=\\"input__field border-black1\\"></div>"
`;

exports[`Figma input component renders default component 1`] = `
"<div class=\\"input\\">
  <!----> <input class=\\"input__field border-black1\\"></div>"
`;
